model_analyzer/analyzer/perf_analyzer.py

```diff
@@ -0,0 +1,192 @@
+            'string-length' : None,
+            'string-data' : None,
+        }
+
+        self._options = {
+            
+            # options
+            '-m' : None,
+            '-x' : None,
+            '-b' : None,
+            '-u' : None,
+            '-i' : None,
+            '-f' : None,
+            '-H' : None
+        }
+
+        self._verbose = {
+            # verbose flags
+            '-v' : None,
+            '-v -v' : None
+        }
+
+    def to_cli_string(self):
+        """Utility function to convert a config into a
+        string of arguments to the perf_analyzer with CLI.
+        """
+        args = ['{} {}'.format(k,v) for k,v in self._options.items() if v]
+        
+        args += [k for k, v in self._verbose.items() if v]
+        
+        args += ['--{}={}'.format(k,v) for k,v in self._args.items() if v]
+
+        return ' '.join(args)
+
+    def __getitem__(self, key):
+        return self._args[key]
+    
+    def __setitem__(self, key, value):
+        input_to_options = {    
+            'model-name' : '-m',
+            'model-version' : '-x',
+            'batch-size' : '-b',
+            'url' : '-u',
+            'protocol' : '-i',
+            'latency-report-file' : '-f',
+            'streaming' : '-H'
+        }
+
+        verbose = {
+            'verbose' : '-v',
+            'extra-verbose' : '-v -v'
+        }
+
+        if key in self._args:
+            self._args[key] = value
+        elif key in input_to_options:
+            self._options[input_to_options[key]] = value
+        elif key in verbose:
+            self._verbose[verbose[key]] = value
+        else:
+            raise Exception("The argument '{}' to the perf_analyzer "
+                             "is not supported by the model analyzer.".format(key))
+
+class PerfAnalyzer:
+    """A wrapper class for the perf_analyzer. This
+    class provides an interface for running workloads 
+    with perf_analyzer
+    """
+    def __init__(self, config):
+        """
+
+        Parameters
+        ----------
+        config : PerfAnalyzerConfig
+            keys are names of arguments to perf_analyzer,
+            values are their values.
+        """
+        self._config = config
+        self._outputs = []
+
+    def run_job(self, sweep_params):
+        """Parses the parameters to sweep over and runs
+           the perf_analyzer once per configuration.
+
+        Parameters
+        ----------
+        sweep_params : dictionary of lists
+            keys are arguments to perf_analyzer, values are 
+            list of argument values to run perf_analyzer with.
+        
+        Returns
+        -------
+        List of tuples of (params, output) where params are the 
+        set of parameters provided for that run and output is the 
+        stdout from the perf_analyzer.
+        """
+
+        # Create a config for each combination of parameters
+        combos = list(product(*tuple(sweep_params.values())))
```

Comment:
This function takes a dict of lists like `{ 'batch-size' : [1,2], 'concurrency-range' : [2,4] }`. This line creates a list of tuples like [(1,2), (1,4), (2,2), (2,4)]. This can then be zipped with the keys from dict to make a config that specifies the parameters for a single run of perf_analyzer.

Maybe the variable should be named differently, or some more commentary is needed?